server/server.go

```diff
@@ -157,7 +216,11 @@ func (s *RpcEndPointServer) handleHealthRequest(respw http.ResponseWriter, req *
 	}
 
 	respw.Header().Set("Content-Type", "application/json")
-	respw.WriteHeader(http.StatusOK)
+	if s.isHealthy {
```

Comment:
don't we need some kind of mutex/atomic for this?

Reply:
in the purest sense, maybe "yes".  but practically, not really.  we don't mind if there's race-condition here (if not this health-check sees the new status, then the next one will).  and it's ok if there is concurrent access (we don't play with pointers here, and we read the value of isHealthy only once in this method).

Reply:
but maybe worthwhile to put this into a comment inside the source-code.  who knows what other functionality will be added here in the future, so the future maintainer should better be aware.